<launch>
	    <!-- vim: set ft=xml noet : -->
   <arg name="plot" default="false"/>

   <node name="controller_x" pkg="pid" type="controller" output="screen">
        <param name="Kp" value="0.10" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
		<param name="upper_limit" value="1" />
		<param name="lower_limit" value="-1" />
		<param name="windup_limit" value="1" />
		<param name="diagnostic_period" value="0.25" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />
        <param name="topic_from_controller" value="effort_x" />
        <param name="topic_from_plant" value="autoland_pose_x" />
        <param name="setpoint_topic" value="autoland_setpoint_x" />
    </node>

    <node name="controller_y" pkg="pid" type="controller" output="screen">
        <param name="Kp" value="0.10" />
        <param name="Ki" value="0.0" />
        <param name="Kd" value="0.0" />
		<param name="upper_limit" value="1" />
		<param name="lower_limit" value="-1" />
		<param name="windup_limit" value="1" />
		<param name="diagnostic_period" value="0.25" />
		<param name="max_loop_frequency" value="100.0" />
		<param name="min_loop_frequency" value="100.0" />
        <param name="topic_from_controller" value="effort_y" />
        <param name="topic_from_plant" value="autoland_pose_y" />
        <param name="setpoint_topic" value="autoland_setpoint_y" />
    </node>

	<group if="$(arg plot)">
		<node name="rqt_plot_x" pkg="rqt_plot" type="rqt_plot" 
				args="/effort_x/data /autoland_pose_x/data /autoland_setpoint_x/data" />
		<node name="rqt_plot_y" pkg="rqt_plot" type="rqt_plot" 
				args="/effort_y/data /autoland_pose_y/data /autoland_setpoint_y/data" />

		<node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

		<node name="piksi" pkg="swiftnav_piksi" type="piksi_node" output="screen">
				<param name="port" value="/dev/ttyUSB0" />
		</node>
	</group>

	</launch>

